test/functional/samples_controller_test.rb

```diff
@@ -7,6 +7,15 @@ class SamplesControllerTest < ActionController::TestCase
   include HtmlHelper
   include GeneralAuthorizationTestCases
 
+  def setup
+    @source_characteristic_isa_tag = FactoryBot.create(:source_characteristic_isa_tag)
+    @sample_characteristic_isa_tag = FactoryBot.create(:sample_characteristic_isa_tag)
+    @other_material_characteristic_isa_tag = FactoryBot.create(:other_material_characteristic_isa_tag)
+    Seek::Samples::BaseType::ALL_TYPES.each do |type|
+      FactoryBot.create(:string_sample_attribute_type, title: "#{type} attibute type", base_type: type)
+    end
+  end
```

Comment:
I don't think we need this for every test, and will create some overhead. Best to put into a method for the tests that require it.